users/script.js

```diff
@@ -588,45 +588,48 @@ applyFilterButton.addEventListener('click', async () => {
   displayLoader();
   const checkedValuesSkills = getCheckedValues('skills-filter');
   const checkedValuesAvailability = getCheckedValues('availability-filter');
+  let page = 0;
+  const size = USER_FETCH_COUNT;
 
-  const queryParams = getFilteredUsersURL(
+  let queryParams = getFilteredUsersURL(
     checkedValuesSkills,
     checkedValuesAvailability,
   );
 
-  const onboarding31DaysFilter =
-    document.getElementById('ONBOARDING31DAYS').checked;
   try {
     let users;
-    if (onboarding31DaysFilter) {
-      let queryParams = getFilteredUsersURL(
-        checkedValuesSkills,
-        checkedValuesAvailability,
-      );
-
+    if (queryParams.includes('&state=ONBOARDING31DAYS')) {
       queryParams = replaceOnboarding31days(queryParams);
-      const usersRequest = await makeApiCall(
-        `${RDS_API_USERS}/search${queryParams}`,
-      );
-      const { users: filteredUsers } = await usersRequest.json();
-      users = filteredUsers;
-    } else {
-      let queryParams = getFilteredUsersURL(
-        checkedValuesSkills,
-        checkedValuesAvailability,
-      );
-      const usersRequest = await makeApiCall(
-        `${RDS_API_USERS}/search${queryParams}`,
-      );
-      const { users: filteredUsers } = await usersRequest.json();
-      users = filteredUsers;
     }
 
+    queryParams += `&page=${page}&size=${size}`;
+
+    queryParams += '&dev=true';
+
+    const usersRequest = await makeApiCall(
+      `${RDS_API_USERS}/search${queryParams}`,
+    );
+    const { users: filteredUsers } = await usersRequest.json();
+    users = filteredUsers;
+
     manipulateQueryParamsToURL(queryParams);
-    // Display the filtered user list
-    showUserList(users);
+    // display the filtered user list with pagination
+    generateUserList(
+      users,
+      true,
+      userListElement,
+      paginationElement,
+      loaderElement,
+      prevBtn,
+    );
   } catch (err) {
-    throw new Error(`User list request failed with error: ${err}`);
+    console.error(`User list request failed with error: ${err}`);
```

Comment:
- use `console.debug`

Reply:
Done. Remove console log and added console debug @yesyash

Reply:
we can use console.error instead of debug right @yesyash ?

Reply:
In the error above i suggested to use console.debug as the error was not valid to be shown to user in console, if we we were logging all console errors for future reference i would have suggested console error